mobile_rewards/src/token_type.rs

```diff
@@ -16,12 +15,17 @@ impl Display for BlockchainTokenTypeV1 {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+#[error("{0} is not a blockchain token type")]
+pub struct NoSuchBlockchainTokenType(i32);
+
 impl TryFrom<i32> for BlockchainTokenTypeV1 {
-    type Error = Error;
-    fn try_from(value: i32) -> Result<Self> {
+    type Error = NoSuchBlockchainTokenType;
```

Comment:
This isn't an alias - it's an associated type

Reply:
I guess what I mean is more whether or not it’s correct to reference the associated type in the scope of the impl as `Self::Error` rather than explicitly by name? Or else why make it an associates type at all? Can you not use the error in that scope otherwise?

Reply:
Neither is more correct, as long as that type `== Self::Error`. I tend to use `Self::Error` to match the trait definition